packages/components/src/components/form/InputButton/InputButton.tsx

```diff
@@ -0,0 +1,177 @@
+import { Row } from '../../Flex/Flex';
+import { IconName } from '../../Icon/Icon';
+import { IconButton } from '../../buttons/IconButton/IconButton';
+import { mapElevationToButtonOnBackground } from '../../buttons/buttonStyleUtils';
+
+const Wrapper = styled.div<{ $elevation: Elevation }>`
+    position: relative;
+    border-radius: ${borders.radii.full};
+    background: ${({ theme, $elevation }) =>
+        mapElevationToButtonOnBackground({ elevation: $elevation, theme, state: 'normal' })};
+`;
+
+const HiddenPlaceholder = styled.div`
+    ${typography.hint};
+    position: absolute;
+    white-space: nowrap;
+    pointer-events: none;
+    visibility: hidden;
+`;
+
+const Input = styled.input<{ $width: number }>`
+    ${typography.hint};
+    height: 100%;
+    border: none;
+    background: transparent;
+    width: ${({ $width }) => $width}px;
+    color: ${({ theme }) => theme.textDefault};
+    padding: 0 ${spacingsPx.lg} 0 ${spacingsPx.xs};
+    box-sizing: content-box;
+
+    &::placeholder {
+        color: ${({ theme }) => theme.textSubdued};
+    }
+`;
+
+const InputWrapper = styled(motion.div)`
+    overflow: hidden;
+`;
+
+export type InputButtonProps = {
+    iconName?: IconName;
+    isExpanded: boolean;
+    value: string;
+    setExpanded: Dispatch<SetStateAction<boolean>>;
+    setValue: Dispatch<SetStateAction<string>>;
+    onChange: (value: string) => void;
+    'data-testid'?: string;
+    placeholder?: string;
+};
+
+export const InputButton = ({
+    placeholder,
+    iconName = 'magnifyingGlass',
+    isExpanded,
+    value,
+    setExpanded,
+    setValue,
+    onChange,
+    'data-testid': dataTest,
+}: InputButtonProps) => {
+    const { elevation } = useElevation();
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const contentRef = useRef<HTMLDivElement | null>(null);
+    const placeholderRef = useRef<HTMLDivElement | null>(null);
+    const [placeholderWidth, setPlaceholderWidth] = useState(0);
+    const [contentWidth, setContentWidth] = useState(0);
+    const isDirty = value !== '';
+
+    useEffect(() => {
+        document.fonts.ready.then(() => {
+            requestAnimationFrame(() => {
+                if (placeholderRef.current) {
+                    setPlaceholderWidth(placeholderRef.current.scrollWidth);
+                }
+            });
+        });
+    }, [placeholder]);
+
+    useEffect(() => {
+        if (contentRef.current) {
+            setContentWidth(contentRef.current.scrollWidth);
+        }
+    }, [placeholderWidth]);
+
+    const toggle = useCallback(() => {
+        if (isDirty) {
+            return;
+        }
+
+        setExpanded(prev => {
+            if (!prev) {
+                inputRef.current?.select();
+            } else {
+                inputRef.current?.blur();
+            }
+
+            return !prev;
+        });
+    }, [setExpanded, isDirty]);
+
+    const clear = useCallback(() => {
+        setValue('');
```

Comment:
Question: why not create a `onClear` callback instead of this imperative `setValue`? TO let the parent component handle the state. Since the parent component handles the state.

Reply:
Oh, I see, you copy the previous API, but I'd change it.